hooks/usePreferences.js

```diff
@@ -0,0 +1,50 @@
+import { gql, useQuery } from '@apollo/client';
+
+export const GET_PREFERENCES = gql`
+  query getPreferences {
+    allPreferences {
+      ...preferencesFragment
+    }
+  }
+  fragment preferencesFragment on Preference {
+    id
+    title
+    summary
+    coverImage {
+      sources {
+        uri
+      }
+    }
+    url
+  }
+`;
+
+export const GET_SUB_PREFERENCES = gql`
+  query getSubPreferences {
+    allSubPreferences {
+      ...subPreferencesFragment
+    }
+  }
+  fragment subPreferencesFragment on SubPreference {
+    id
+    title
+    coverImage {
+      sources {
+        uri
+      }
+    }
+  }
+`;
+
+function usePreferences(options = {}) {
+  const queryPreferences = useQuery(GET_PREFERENCES, options);
+  const querySubPreferences = useQuery(GET_SUB_PREFERENCES, options);
+  return {
+    preferences: queryPreferences?.data?.allPreferences || [],
+    subpreferences: querySubPreferences?.data?.allSubPreferences || [],
+    ...queryPreferences,
+    ...querySubPreferences,
+  };
+}
```

Comment:
So I'm on the fence re: the "should we split these?" question.

For now, I don't see a strong case for splitting, since we kinda need to know the full data set for both whenever we need to know one or the other.

And it's a bit of a moot point with Apollo client caching. (Assuming it's still `cache-and-network` by default).

The weird thing may be that the `loading`, `error`, and `data` props from the `querySubPreferences` would overwrite the `queryPreferences` values, right?
So consumers of this hook may get tripped up thinking `loading` refers to the combined loading state, etc?

At same time, separate hooks sounds easily workable and clean, as well.

Reply:
Oops, good catch on the spread overwriting each other. 👍